packages/server-wallet/src/protocols/defunder.ts

```diff
@@ -0,0 +1,104 @@
+import {unreachable} from '@statechannels/wallet-core';
+import {Transaction} from 'objection';
+import {Logger} from 'pino';
+
+import {ChainServiceInterface} from '../chain-service';
+import {AdjudicatorStatusModel} from '../models/adjudicator-status';
+import {ChainServiceRequest} from '../models/chain-service-request';
+import {Channel} from '../models/channel';
+import {LedgerRequest} from '../models/ledger-request';
+import {Store} from '../wallet/store';
+
+export type DefunderResult = {isChannelDefunded: boolean; didSubmitTransaction: boolean};
+
+export class Defunder {
+  constructor(
+    private store: Store,
+    private chainService: ChainServiceInterface,
+    private logger: Logger,
+    private timingMetrics = false
+  ) {}
+
+  public static create(
+    store: Store,
+    chainService: ChainServiceInterface,
+    logger: Logger,
+    timingMetrics = false
+  ): Defunder {
+    return new Defunder(store, chainService, logger, timingMetrics);
+  }
+
+  public async isChannelDefunded(channel: Channel, tx: Transaction): Promise<DefunderResult> {
```

Comment:
I think I'd prefer `crank` as `isChannelDefunded` sounds more like a check than a function that is doing work.

Reply:
Agreed! 1a2bc6705